packages/client/src/pages/Game.tsx

```diff
@@ -22,51 +23,32 @@ const SETTEBELLO = {
 export const Game = ({ gameState, gameScore }: { gameState: GameState; gameScore?: Score[] }) => {
   const [activePlayerCard, togglePlayerActiveCard] = useState<string | null>(null);
   const [activeCardsOnTable, toggleActiveCardsOnTable] = useState<string[]>([]);
-  const [movingCards, toggleMovingCards] = useState<string[]>([]);
   const { username } = useUserData();
   const { roomName } = useParams<{ roomName: string }>();
 
   const { activePlayer, deck, table, players } = gameState;
 
   // TODO figure out what to do when more than 2 players
-  const [opponent] = useMemo(() => players.filter((player) => player.username !== username), [players, username]);
-  const [player] = useMemo(() => players.filter((player) => player.username === username), [players, username]);
+  const { player, opponent } = useMemo(
+    () => Object.fromEntries(players.map((p) => [p.username === username ? 'player' : 'opponent', p])),
+    [players, username],
+  );
   useEffect(() => {
-    let timer: NodeJS.Timeout;
-
     if (activePlayerCard && activeCardsOnTable) {
       const { value: playerCardNumber } = fromCardKey(activePlayerCard);
       const tableCardsSum = sum(activeCardsOnTable.map((c) => fromCardKey(c).value));
       if (playerCardNumber === tableCardsSum) {
-        toggleMovingCards([activePlayerCard, ...activeCardsOnTable]);
-        timer = setTimeout(() => {
-          console.log('Emit: Cards are going to playerCaptured');
-          gameIO.emit(GameEvent.UpdateState, roomName, {
-            ...gameState,
-            activePlayer: opponent.username,
-            players: [
-              opponent,
-              {
-                ...player,
-                hand: player.hand.filter((c) => cardKey(c) !== activePlayerCard),
-                captured: [
-                  ...player.captured,
-                  fromCardKey(activePlayerCard),
-                  ...activeCardsOnTable.map((c) => fromCardKey(c)),
-                ],
-              },
-            ],
-            table: gameState.table.filter((c) => !activeCardsOnTable.includes(cardKey(c))),
-            latestCaptured: username,
-          });
-          togglePlayerActiveCard(null);
-          toggleActiveCardsOnTable([]);
-          toggleMovingCards([]);
-        }, 600);
+        gameIO.emit(GameEvent.PlayerAction, roomName, {
+          action: PlayerActionType.Capture,
+          playerName: player.username,
+          card: activePlayerCard,
+          cardsFromTable: activeCardsOnTable,
```

Comment:
```suggestion
          tableCards: activeCardsOnTable,
```
Or `cardsOnTable` if you want to keep the same names.